pkg/cmd/opts/install.go

```diff
@@ -223,35 +223,61 @@ func (o *CommonOptions) InstallGlooctl() error {
 }
 
 // InstallKustomize installs kustomize
-func (o *CommonOptions) InstallKustomize() error {
+func (o *CommonOptions) InstallKustomize() (err error) {
 	if runtime.GOOS == "darwin" && !o.NoBrew {
 		return o.RunCommand("brew", "install", "kustomize")
 	}
+
 	binDir, err := util.JXBinLocation()
 	if err != nil {
 		return err
 	}
+
 	fileName, flag, err := packages.ShouldInstallBinary("kustomize")
 	if err != nil || !flag {
 		return err
 	}
 
-	latestVersion, err := util.GetLatestVersionFromGitHub("kubernetes-sigs", "kustomize")
+	// get the stable jx supported version of kustomize to be install
+	versionResolver, err := o.GetVersionResolver()
 	if err != nil {
-		return fmt.Errorf("unable to get latest version for github.com/%s/%s %v", "kubernetes-sigs", "kustomize", err)
+		log.Logger().Warnf("Unable to get version resolver for jenkins-x-versions %s", err)
 	}
 
-	clientURL := fmt.Sprintf("https://github.com/kubernetes-sigs/kustomize/releases/download/v%v/kustomize_%s_%s_%s", latestVersion, latestVersion, runtime.GOOS, runtime.GOARCH)
-	fullPath := filepath.Join(binDir, fileName)
-	tmpFile := fullPath + ".tmp"
-	err = packages.DownloadFile(clientURL, tmpFile)
+	stableVersion, err := versionResolver.StableVersion(versionstream.KindPackage, "kustomize")
+	if err != nil {
+		return fmt.Errorf("Unable to get stable version from the jenkins-x-versions for github.com/%s/%s %v ", "kubernetes-sigs", "kustomize", err)
+	}
+
+	clientURL := fmt.Sprintf("https://github.com/kubernetes-sigs/kustomize/releases/download/kustomize%%2Fv%s/kustomize_v%s_%s_%s.tar.gz", stableVersion.Version, stableVersion.Version, runtime.GOOS, runtime.GOARCH)
+	tmpDir := filepath.Join(binDir, "kustomize.tmp")
+	err = os.MkdirAll(tmpDir, util.DefaultWritePermissions)
 	if err != nil {
 		return err
 	}
-	err = util.RenameFile(tmpFile, fullPath)
+	fullPath := filepath.Join(binDir, "kustomize")
+	tarFile := filepath.Join(tmpDir, fileName+".tar.gz")
+	err = packages.DownloadFile(clientURL, tarFile)
+	if err != nil {
+		return err
+	}
+	err = util.UnTargz(tarFile, tmpDir, []string{"kustomize"})
+	if err != nil {
+		return err
+	}
+	err = os.Remove(tarFile)
+	if err != nil {
+		return err
+	}
+	err = os.Rename(filepath.Join(tmpDir, "kustomize"), fullPath)
 	if err != nil {
 		return err
 	}
+	err = os.RemoveAll(tmpDir)
```

Comment:
This has to happen in a _defer_ clause which you register after the _MkdirAll_ call. Maybe even before.